spec/job/adapters/slurm_spec.rb

```diff
@@ -1106,6 +1106,40 @@ def job_info(opts = {})
     end
   end
 
+  describe "setting submit_host" do 
+    let (:script) { OodCore::Job::Script.new(content: "echo'hi'") } 
+
+    context "when calling without submit_host" do 
+      it "uses the correct command" do 
+        batch = OodCore::Job::Adapters::Slurm::Batch.new(cluster: "owens.osc.edu", conf: "/etc/slurm/conf/", bin: nil, bin_overrides: {}, submit_host: "")
+        allow(Open3).to receive(:capture3).and_return(["job.123", "", double("success?" => true)])
+
+        OodCore::Job::Adapters::Slurm.new(slurm: batch).submit script
+        expect(Open3).to have_received(:capture3).with(anything, "sbatch", '--export', 'NONE', '--parsable', '-M', 'owens.osc.edu', any_args)
+      end
+    end
+
+    context "when calling with submit_host & strict_host_checking not specified" do 
+      it "uses ssh wrapper & host checking defaults to true" do 
+        batch = OodCore::Job::Adapters::Slurm::Batch.new(cluster: "owens.osc.edu", conf: "/etc/slurm/conf/", bin: nil, bin_overrides: {}, submit_host: "owens.osc.edu")
+        allow(Open3).to receive(:capture3).and_return(["job.123", "", double("success?" => true)])
+
+        OodCore::Job::Adapters::Slurm.new(slurm: batch).submit script
+        expect(Open3).to have_received(:capture3).with(anything, 'ssh', '-o', 'BatchMode=yes', '-o', 'UserKnownHostsFile=/dev/null', '-o', 'StrictHostKeyChecking=yes', 'owens.osc.edu', 'sbatch --export NONE --parsable -M owens.osc.edu', any_args)
```

Comment:
Why do we do

    'owens.osc.edu', 'sbatch --export NONE --parsable -M owens.osc.edu'

instead of

    'owens.osc.edu', 'sbatch', '--export', 'NONE', '--parsable', '-M', 'owens.osc.edu'

Reply:
In bash, something like `'touch foo.txt; echo foo > foo.txt'` would only work with quotations. However ruby's open3 it works as an array instead as one string. (Plus we aren't using multiple commands)
Will change back to an array for args.